ansible/plugins/callback/datadog_callback.py

```diff
@@ -0,0 +1,315 @@
+                metric="ansible.{0}".format(metric),
+                points=value,
+                tags=tags,
+                host=host,
+            )
+        except Exception as e:
+            # We don't want Ansible to fail on an API error
+            print('Couldn\'t send metric "{0}" to Datadog'.format(metric))
+            print(e)
+
+    # Start timer to measure playbook running time
+    def start_timer(self):
+        self._start_time = time.time()
+
+    # Get the time elapsed since the timer was started
+    def get_elapsed_time(self):
+        return time.time() - self._start_time
+
+    # Default tags sent with events and metrics
+    @property
+    def default_tags(self):
+        return ['playbook:{0}'.format(self._playbook_name)]
+
+    @staticmethod
+    def pluralize(number, noun):
+        if number == 1:
+            return "{0} {1}".format(number, noun)
+
+        return "{0} {1}s".format(number, noun)
+
+    # format helper for event_text
+    @staticmethod
+    def format_result(res):
+        msg = "$$$\n{0}\n$$$\n".format(res['msg']) if res.get('msg') else ""
+        module_name = 'undefined'
+
+        if res.get('censored'):
+            event_text = res.get('censored')
+        elif not res.get('invocation'):
+            event_text = msg
+        else:
+            invocation = res['invocation']
+            module_name = invocation.get('module_name', 'undefined')
+            event_text = "$$$\n{0}[{1}]\n$$$\n".format(module_name, invocation.get('module_args', ''))
+            event_text += msg
+            if 'module_stdout' in res:
+                # On Ansible v2, details on internal failures of modules are not reported in the `msg`,
+                # so we have to extract the info differently
+                event_text += "$$$\n{0}\n{1}\n$$$\n".format(
+                    res.get('module_stdout', ''), res.get('module_stderr', ''))
+
+        module_name_tag = 'module:{0}'.format(module_name)
+
+        return event_text, module_name_tag
+
+    ### Ansible callbacks ###
+    def runner_on_failed(self, host, res, ignore_errors=False):
+        # don't post anything if user asked to ignore errors
+        if ignore_errors:
+            return
+
+        event_text, module_name_tag = self.format_result(res)
+        self.send_task_event(
+            'Ansible task failed on "{0}"'.format(host),
+            alert_type='error',
+            text=event_text,
+            tags=[module_name_tag],
+            host=host,
+        )
+
+    def runner_on_ok(self, host, res):
+        # Only send an event when the task has changed on the host
+        if res.get('changed'):
+            event_text, module_name_tag = self.format_result(res)
+            self.send_task_event(
+                'Ansible task changed on "{0}"'.format(host),
+                alert_type='success',
+                text=event_text,
+                tags=[module_name_tag],
+                host=host,
+            )
```

Comment:
these task events don't seem very useful since they don't include any info about the task at all

Reply:
Datadog -> events -> Search (Ansible) Give lot of information . (Who ran which playbook on which server and what changed)

Reply:
Maybe I'm not looking in the right place but I don't see that much info:
![image](https://user-images.githubusercontent.com/249606/36665303-d2084ac8-1aef-11e8-8df7-e2f882bb4dba.png)

Reply:
Yes, you are right... That's the same. Sorry,  I think I got confused with some error that I saw on my terminal. 
<img width="813" alt="screen shot 2018-02-26 at 4 07 11 pm" src="https://user-images.githubusercontent.com/11015404/36666040-52abc960-1b0f-11e8-90bf-dccae871445c.png">

Reply:
It would be possible to add more info in if we wanted to. There are plenty of examples here: https://github.com/ansible/ansible/tree/devel/lib/ansible/plugins/callback

I think of most use would be:
* date time
* user running playbook
* playbook that was run
* other options (limit, tags etc)
* any changes that were made (optional)